services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/FileScanService.kt

```diff
@@ -0,0 +1,86 @@
+package fi.hel.haitaton.hanke.security
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import fi.hel.haitaton.hanke.toJsonString
+import java.time.Duration.ofSeconds
+import mu.KotlinLogging
+import org.apache.commons.lang3.BooleanUtils.isNotFalse
+import org.springframework.core.io.ByteArrayResource
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.MediaType.APPLICATION_PDF
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.BodyInserters.fromMultipartData
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+
+private val logger = KotlinLogging.logger {}
+
+private const val FORM_KEY = "FILES"
+
+@Service
+class FileScanService(
+    private val fileScanClient: WebClient,
+) {
+
+    fun scanFiles(files: Set<Pair<String, ByteArray>>): FileScanResponse {
```

Comment:
FileScanResponse.success is always true when this function returns, right?  And if `hasInfected` is a function, it could be an extension function on FileScanData.

Then again FileScanData only has one field, result. So it can be unwrapped right away.

So if we have
```
fun List<FileResult>.hasInfected(): Boolean = any { isNotFalse(it.isInfected) }
```
defined somewhere, I think we can return just `List<FileResult>` from this class.

Reply:
Fixed as suggested.